### DISCUSSION TITLE
pseudo-class (styled-components) is not consistent

### DISCUSSION THREAD

**Discussion by jonatsuji** [neuefische/questions#439] [Web] [answered]
hey everyone, I hope I can explain my problem in a good way.

I added styled components for a search bar and in there I have a pseudo class for the place holder. The placeholder must disappear when the user clicked in the search bar. I have two pages where this works but on one page, I implemented the same searchBar component but the placeholder doesn't disappear when it should:

<img width="418" alt="Screenshot 2023-01-05 at 14 35 50" src="https://user-images.githubusercontent.com/115465230/210793709-ffebedb3-6617-451d-8bac-f9ed0448099f.png">
<img width="201" alt="Screenshot 2023-01-05 at 14 35 56" src="https://user-images.githubusercontent.com/115465230/210793767-187f3914-14a5-4f8d-962b-82487fa47a93.png">
<img width="216" alt="Screenshot 2023-01-05 at 14 36 09" src="https://user-images.githubusercontent.com/115465230/210793785-105a13eb-4e3d-4d6e-a3d0-df32f2ee4f8a.png">

Code:

<img width="438" alt="Screenshot 2023-01-05 at 14 43 19" src="https://user-images.githubusercontent.com/115465230/210794074-f96cec86-e5bb-494b-8aff-2e0087ba59ab.png">

Here you find the current [PR](https://github.com/jonatsuji/capstone-project/pull/6) to have a look into the code.

**Comment by MerleMak** [comment, marked as answer]
Hi Jonathan!
Do I understand correctly that you want the placeholder to disappear when focussing on the input?
with `placeholder-shown` you are actually styling the input where the placeholder in being shown. Check [this link for further information](https://css-tricks.com/almanac/selectors/p/placeholder-shown/#aa-the-difference-between-placeholder-shown-and-placeholder)
So maybe try to work with `::placeholder` instead and let me know if it works :)

**Reply by jonatsuji** [reply]
Hi @MerleMak, I found the issue. It worked before with "placeholder-shown" so I was confused but I found that I implemented the SearchBar component twice, so the placeholder was always shown from the second searchbar.

Thanks for your advice anyway!!
